packages/DataTable/src/context.js

```diff
@@ -86,8 +92,74 @@ function useDispatch() {
   return React.useContext(TableDispatchContext);
 }
 
+function useData() {
```

Comment:
I think this is enough to have their own file which is a hook maybe simple extract it outside on this file context and put it next to it. as `useData.js`?